src/main.rs

```diff
@@ -76,6 +78,15 @@ struct Wstunnel {
         default_value = "INFO"
     )]
     log_lvl: String,
+
+    /// Set the listen address for the prometheus metrics exporter.
+    #[arg(
+        long,
+        global = true,
+        verbatim_doc_comment,
+        default_value = None,
+    )]
+    metrics_provider_address: Option<String>,
```

Comment:
You can directly set an [IpAddress in](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) instead of a string